### PR TITLE
Use `window_size_bytes: auto` to specify automatic windowing

### PR DESCRIPTION
**Pull request by jeffkinnison**
This updates `RayDataset`, `RayDatasetManager`, and `RayBackend` to accept `"auto"` as a valid input for `window_size_bytes`, and consolidates `RayDataset` creation logic to handle auto-windowing.

### CODE REVIEW

File: ludwig/data/dataset/ray.py [hunk 1]
```diff
@@ -60,20 +60,15 @@ def read_remote_parquet(path: str):
 
 @DeveloperAPI
 class RayDataset(Dataset):
-    """Wrapper around ray.data.Dataset.
-
-    Attributes:
-        auto_window: If True and the dataset is larger than available memory,
-            automatically set window size to `<available memory> // 5`.
-    """
+    """Wrapper around ray.data.Dataset."""
```
Comment (arnavgarg1) [inline ludwig/data/dataset/ray.py:63]: nit: While we're updating the doc strings, should we update it to all of the inputs and their types for this class? toally ok to skip as well

Comment (jeffkinnison) [reply ludwig/data/dataset/ray.py:63]: Ah, good catch. I meant to add `window_size_bytes` back in and fill in the rest of the docstring.

File: ludwig/data/dataset/ray.py [hunk 2]
```diff
@@ -82,16 +77,21 @@ def __init__(
...
+    def get_window_size_bytes(self, window_size_bytes: Optional[Union[int, Literal["auto"]]] = None) -> int:
+        """Return this dataset's window size in bytes, or translate auto-windowing into bytes."""
+        # By default, set to -1 so that an infinite window size
+        # will be used which effectively results in bulk data ingestion
+        window_size = -1
 
-        # If the user does not supply a window size and the dataset is large,
+        # If user has specified a window size, use it as-is.
+        if isinstance(window_size_bytes, int):
+            window_size = window_size_bytes
```
[the remark this answers is not in the document]
Comment (jeffkinnison) [reply ludwig/data/dataset/ray.py:90]: I'm generally a fan of a single exit point for a function unless some serious computation happens, but happy to refactor.

[the remark this answers is not in the document]
Comment (arnavgarg1) [reply ludwig/data/dataset/ray.py:90]: That's fair, can leave this as is

File: tests/integration_tests/test_ray.py [hunk 3]
```diff
@@ -1015,15 +1014,14 @@ def create_dataset_pipeline(
             "output_features": [{"name": "out_column", "type": "binary"}],
             TRAINER: {"epochs": 1, BATCH_SIZE: 128},
         }
-        backend_config = {**RAY_BACKEND_CONFIG}
+        backend_config = copy.deepcopy(RAY_BACKEND_CONFIG)
+        backend_config["loader"] = {"window_size_bytes": window_size_bytes}
```
Comment (arnavgarg1) [inline tests/integration_tests/test_ray.py:1018]: Just curious - are we always guaranteed to have the `loader` key in the backend config? If not, do we need to add a default value for this in the schema (possibly `None`)?

File: ludwig/data/dataset/ray.py (hunk 1 above)
Comment (jeffkinnison) [reply ludwig/data/dataset/ray.py]: Added in [db4218b](https://github.com/ludwig-ai/ludwig/pull/3076/commits/db4218bbb5404fb45e3a7795b6177eea0f05484e)

File: ludwig/data/dataset/ray.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeffkinnison) [reply ludwig/data/dataset/ray.py]: Revised in [db4218b](https://github.com/ludwig-ai/ludwig/pull/3076/commits/db4218bbb5404fb45e3a7795b6177eea0f05484e)

File: tests/integration_tests/test_ray.py (hunk 3 above)
Comment (jeffkinnison) [reply tests/integration_tests/test_ray.py:1018]: Under the hood, this update uses `backend._data_loader_kwargs` which the `RayBackend` initializer sets with `self._data_loader_kwargs = loader or {}`. We don't actually rely on the existence of `loader`, but rather the backend's handling of its config.

**Review by tgaddair — APPROVED** [review]

File: ludwig/data/dataset/ray.py [hunk 4]
```diff
@@ -82,16 +86,17 @@ def __init__(
         self.data_hdf5_fp = training_set_metadata.get(DATA_TRAIN_HDF5_FP)
         self.data_parquet_fp = training_set_metadata.get(DATA_TRAIN_PARQUET_FP)
         self._processed_data_fp = df if isinstance(df, str) else None
-        self.auto_window = auto_window
+        self.window_size_bytes = self.get_window_size_bytes(window_size_bytes)
 
-    def get_window_size_bytes(self, window_size_bytes: Optional[int] = None) -> int:
-        # If user has specified a window size, use it as is
-        if window_size_bytes:
+    def get_window_size_bytes(self, window_size_bytes: Optional[Union[int, Literal["auto"]]] = None) -> int:
```
Comment (arnavgarg1) [inline ludwig/data/dataset/ray.py:91]: Really nice use of literal, we should use that more in Ludwig

**Review by arnavgarg1 — APPROVED** [review]
Great work! Thanks for cleaning up the implementation/API

**Pull request merged**